src/Api.Rest/Common/Authentication/NonInteractiveAuthenticationHandler.cs

```diff
@@ -0,0 +1,304 @@
+	/// method must execute the refresh token consumer on an internally stored OIDC refresh token. The consumer returns
+	/// a new refresh token that must then replace the internally stored refresh token for any future calls. If the input refresh token of
+	/// the consumer can be reused again after this operation, the consumer may return this input refresh token instead of a new token.
+	/// If no internally stored refresh token is available for any reason, the processor method should not execute the consumer at all.
+	/// The return value of the processor must indicate whether the consumer was executed (<c>true</c>), or not executed (<c>false</c>).
+	/// Using a processor instead of a static refresh token enables refresh token rotation.
+	/// </summary>
+	/// <param name="refreshTokenProcessor">A method to process and potentially rotate the existing refresh token.</param>
+	/// <returns>The new instance.</returns>
+	public static NonInteractiveAuthenticationHandler OIDC( RefreshTokenProcessorAsync refreshTokenProcessor )
+	{
+		return new NonInteractiveAuthenticationHandler(
+			AuthenticationMode.OAuth,
+			refreshTokenProcessor: refreshTokenProcessor );
+	}
+
+	/// <summary>
+	/// Creates a new <see cref="NonInteractiveAuthenticationHandler"/> instance which implements Single Sign-on authentication based
+	/// on Kerberos or NTLM authentication.
+	/// </summary>
+	/// <returns>The new instance.</returns>
+	public static NonInteractiveAuthenticationHandler WindowsSSO()
+	{
+		return new NonInteractiveAuthenticationHandler( AuthenticationMode.Windows );
+	}
+
+	/// <summary>
+	/// Creates a new <see cref="NonInteractiveAuthenticationHandler"/> instance which implements username/password authentication based
+	/// on Kerberos or NTLM authentication.
+	/// </summary>
+	/// <param name="username">The username.</param>
+	/// <param name="password">The password.</param>
+	/// <returns>The new instance.</returns>
+	public static NonInteractiveAuthenticationHandler Windows( string username, string password )
+	{
+		return new NonInteractiveAuthenticationHandler( AuthenticationMode.Windows, username, password );
+	}
+
+	/// <summary>
+	/// Creates a new <see cref="NonInteractiveAuthenticationHandler"/> instance which implements username/password authentication based
+	/// on Basic authentication.
+	/// </summary>
+	/// <param name="username">The username.</param>
+	/// <param name="password">The password.</param>
+	/// <returns>The new instance.</returns>
+	public static NonInteractiveAuthenticationHandler Basic( string username, string password )
+	{
+		return new NonInteractiveAuthenticationHandler( AuthenticationMode.NoneOrBasic, username, password );
+	}
+
+	/// <summary>
+	/// Creates a new <see cref="NonInteractiveAuthenticationHandler"/> instance which implements certificate authentication based
+	/// on https client certificates.

```

Comment:
Maybe I would not call it an _https_ client certificate, but only client certificate. Https and client certificates are of course the same from a technical point of view, but here we want to distinguish by the use case. Also in the other certificate method below.

Reply:
Ok